File: src/main/java/io/dddrive/property/model/ReferenceSetProperty.java
package io.dddrive.property.model;

import java.util.Collection;
import java.util.Set;

import io.dddrive.ddd.model.Aggregate;

public interface ReferenceSetProperty<A extends Aggregate> extends CollectionProperty<A> {

	Set<Integer> getItems();

	boolean hasItem(Integer aggregateId);

	void clearItems();

	void addItem(Integer aggregateId);

	void removeItem(Integer aggregateId);

	void loadReferences(Collection<? extends AggregatePartItem<?>> items);

}

File: fm-dddrive/src/main/java/io/dddrive/property/model/ReferenceSetProperty.kt
package io.dddrive.property.model

import io.dddrive.ddd.model.Aggregate

/**
 * Property that holds a set of aggregate references (by ID).
 *
 * Implements [Set] interface so it can be used directly as a set in consumer code.
 * Note: The set contains IDs, not the aggregates themselves.
 */
interface ReferenceSetProperty<A : Aggregate> :
	Property<A>,
	Collection<Any> {

	fun has(aggregateId: Any): Boolean

	fun clear()

	fun add(aggregateId: Any)

	fun remove(aggregateId: Any)

}
